test/__snapshots__/runAll.spec.js.snap

```diff
@@ -29,17 +29,7 @@ LOG Running tasks for *.js [completed]
 LOG Running linters... [completed]"
 `;
 
-exports[`runAll should resolve the promise with no files 1`] = `
-"
-LOG Stashing changes... [started]
-LOG Stashing changes... [skipped]
-LOG → No partially staged files found...
-LOG Running linters... [started]
-LOG Running tasks for *.js [started]
-LOG Running tasks for *.js [skipped]
-LOG → No staged files match *.js
-LOG Running linters... [completed]"
-`;
+exports[`runAll should resolve the promise with no files 1`] = `""`;
```

Comment:
This doesn’t look correct to me. As a user I’d still like to get some feedback on why it’s not executed. The output here was correct and we should match it since both conditions are true. I think what you want to achieve is a `skip` function on the stashing step that should check and return with “No staged files match any of provided globs” or similar. WYT?

Reply:
My logic was to skip executing *any* kind of command if no staged files matched the provided globs. If we prepend more tasks in the future like the one that stashes partial changes, I'd rather *everything* be guaranteed to be skipped.

But I'm happy to move the code there if you feel if it makes more sense.

Reply:
I’m more concerned about the UX here as the silence could be frustrating for users. What do you think?

Reply:
Yeah that's a fair point.

In other programming language communities, like Go, generally you only log if things were modified/changed. If nothing was processed or changed then nothing is logged. This is why I felt it was OK to log nothing.

In saying all this... I'm not sure this makes sense for a tool that is generally executed by humans not continuous integration machines. This behaviour might feel incorrect within the JavaScript community too.

So... maybe we can just log what you suggested and exit early so that the snapshot outputs:
```
LOG No linters executed. No staged files match any of provided globs in "lint-staged.linters"."
```

Reply:
That sounds good to me. So you still think we should not use Listr for this and exit earlier?

Reply:
I do.
My expectation is that the program would exit as soon as possible once it realizes it shouldn't process anything. Is that ok with you?

Reply:
Sounds good to me! Let’s do this

Reply:
Awesome :)
I'll most likely have time to get onto this in 2-3 days!